File: DecisionTreeClassifier.js
module.exports = function(RED){
	function DecisionTreeClassifierNode(config){
		const path = require('path')
		const utils = require('./utils/utils')

		var node = this;

		//set configurations
		node.file = __dirname + '/DecisionTreeClassifier.py'
		node.config = {
			target: config.target || 'y',
			savePath: config.savePath || "DecisionTreeClassifier.sav",
			patience: parseInt(config.patience) || 10,
			max_depth: parseInt(config.max_depth) || 5,
			min_child_samples: parseInt(config.min_child_samples) || 20,
			confidence: Number(config.confidence) || 1e-05,
			tie_threshold: Number(config.tie_threshold) || 0.05
		}
		utils.run(RED, node, config)
	}
	RED.nodes.registerType("DecisionTreeClassifier", DecisionTreeClassifierNode)
}

File: DecisionTreeClassifier.py
import json
import pickle
from creme import tree

config = json.loads(input())

savePath = config['savePath']
target = config['target']
patience = float(config['patience'])
max_depth = config['max_depth']
min_child_samples = config['min_child_samples']
confidence = config['confidence']
tie_threshold = config['tie_threshold']
	

DTClassifier = tree.DecisionTreeClassifier('gini', patience, max_depth, min_child_samples, confidence, tie_threshold)

while True:

	#wait request
	data = input()

	Xi = json.loads(data)
	y = Xi.pop(target, None)
	print(Xi)
	print(y)
	
	model = DTClassifier.fit_one(Xi, y)
	pickle.dump(model, open(savePath, 'wb'))